subsys/mgmt/mcumgr/serial_util.c

```diff
@@ -241,8 +239,14 @@ int mcumgr_serial_tx_frame(const uint8_t *data, bool first, int len,
 
 	if (reminder == 0 || reminder == 1) {
 		to_process -= 1;
-	} else if (reminder >= 2) {
-		last = true;
+
+		if (tx_data->flags & SMP_TRANSPORT_OUT_LAST_PACKET) {
+			/* This is the last frame but the checksum cannot be added, remove the
+			 * last packet flag until the function runs again with the final piece
+			 * of data and place the checksum there
+			 */
+			tx_data->flags ^= SMP_TRANSPORT_OUT_LAST_PACKET;
+		}
```

Comment:
No need for branches here, just do the `tx-data->flags &= ~SMP_TRANSPORT_OUT_LAST_PACKET`.

Reply:
Fixed